eq_translations/utils.py

```diff
@@ -50,6 +52,34 @@ def find_pointers_to(input_data, search_key):
     return root_pointers + [f"{pointer}/{search_key}" for pointer in pointer_iterator]
 
 
+def get_parent_schema_object(input_data, json_pointer, parent_property):
+    """
+    Get the parent schema object identified by parent_property in the json pointer.
+    If the parent schema object is an array then the matching array item is returned.
+    :param input_data: the input data to search
+    :param json_pointer: the pointer being searched
+    :param parent_property: the parent property to search for
+    :return: schema object identified by parent_property
+    """
+    pointer_parts = json_pointer.split("/")
+    pointer_index = pointer_parts.index(parent_property)
+    parent_pointer = "/".join(pointer_parts[: pointer_index + 1])
+    schema_object = resolve_pointer(input_data, parent_pointer)
+    if isinstance(schema_object, list):
+        return schema_object[int(pointer_parts[pointer_index + 1])]
+    return schema_object
```

Comment:
If these were jsonpath objects, couldn't we make use of the available context that it provides?

Reply:
I did look at the documentation for JSONPath and although it implied it could be used for this, there were no examples. Have you done this somewhere else?

Reply:
I've done it in my branch for validator within QuestionnaireSchema - get_block_key_context

Reply:
Does that help with finding any parent property, or just the immediate parent?